File: android/app/src/main/java/com/amiya/pet/core/sync/SyncModels.kt
package com.amiya.pet.core.sync

import org.json.JSONObject

/**
 * 局域网互联设备信息
 */
data class DeviceInfo(
    val deviceId: String,
    val deviceName: String,
    val deviceType: String, // "pc" or "android"
    val ip: String,
    val httpPort: Int = 23334,
    val version: String = "1.9.0",
    val paired: Boolean = false
) {
    fun toJson(): JSONObject {
        return JSONObject().apply {
            put("device_id", deviceId)
            put("device_name", deviceName)
            put("device_type", deviceType)
            put("ip", ip)
            put("http_port", httpPort)
            put("version", version)
            put("paired", paired)
        }
    }

    companion object {
        fun fromJson(json: JSONObject, fallbackIp: String = ""): DeviceInfo {
            return DeviceInfo(
                deviceId = json.optString("device_id", ""),
                deviceName = json.optString("device_name", "未知设备"),
                deviceType = json.optString("device_type", "pc"),
                ip = json.optString("ip", fallbackIp),
                httpPort = json.optInt("http_port", 23333),
                version = json.optString("version", "1.0.0"),
                paired = json.optBoolean("paired", false)
            )
        }
    }
}

/**
 * 等待用户显式确认的配对请求
 */
data class PairRequest(
    val requestId: String,
    val clientId: String,
    val clientName: String,
    val clientIp: String,
    val clientPort: Int,
    val pin: String
)

/**
 * 战术快传投递内容
 */
data class TacticalDrop(
    val text: String,
    val title: String = "来自博士的终端",
    val timestamp: Long = System.currentTimeMillis()
)

/**
 * 学业数据同步结果汇报
 */
data class SyncReport(
    val success: Boolean,
    val message: String,
    val courseCount: Int = 0,
    val examCount: Int = 0,
    val noteCount: Int = 0
)

File: ios/AmiyaPet/Core/Sync/SyncModels.swift
import Foundation

public struct DeviceInfo: Codable, Identifiable, Hashable {
    public var id: String { deviceId }
    public var deviceId: String
    public var deviceName: String
    public var deviceType: String // "pc", "android", "ios"
    public var ip: String
    public var httpPort: Int
    public var version: String
    public var paired: Bool

    public init(
        deviceId: String,
        deviceName: String,
        deviceType: String = "ios",
        ip: String,
        httpPort: Int = 23334,
        version: String = "1.9.3",
        paired: Bool = false
    ) {
        self.deviceId = deviceId
        self.deviceName = deviceName
        self.deviceType = deviceType
        self.ip = ip
        self.httpPort = httpPort
        self.version = version
        self.paired = paired
    }

    enum CodingKeys: String, CodingKey {
        case deviceId = "device_id"
        case deviceName = "device_name"
        case deviceType = "device_type"
        case ip
        case httpPort = "http_port"
        case version
        case paired
    }
}

public struct TacticalDrop: Codable {
    public let text: String
    public let title: String
    public let timestamp: Int64

    public init(text: String, title: String = "战术快传", timestamp: Int64 = Int64(Date().timeIntervalSince1970 * 1000)) {
        self.text = text
        self.title = title
        self.timestamp = timestamp
    }
}
